src/wallet/components/ConnectWallet.tsx

```diff
@@ -24,12 +25,16 @@ export function ConnectWallet({
   // but for now we will keep it for backward compatibility.
   text = 'Connect Wallet',
   withWalletAggregator = false,
+  onConnect,
 }: ConnectWalletReact) {
   // Core Hooks
   const { isOpen, setIsOpen } = useWalletContext();
   const { address: accountAddress, status } = useAccount();
   const { connectors, connect, status: connectStatus } = useConnect();
 
+  // State
+  const [hasClickedConnect, setHasClickedConnect] = React.useState(false);
```

Comment:
Ideally we'd be able to avoid this piece of state, but there's no other way to detect whether `status === 'connected'` is true because:

1.  the user has just signed in OR
2. the user has refreshed the page